daceml/autodiff/pytorch.py

```diff
@@ -0,0 +1,200 @@
+import logging
+from typing import Type
+import itertools
+from collections import OrderedDict
+
+import torch
+
+import dace
+from dace import data as dt
+
+from daceml.autodiff.backward_pass_generator import BackwardPassGenerator
+from daceml.autodiff.base_abc import AutoDiffException
+from daceml.onnx.converters import clean_onnx_name
+from daceml.onnx.onnx_importer import create_output_array, ONNXModel
+
+log = logging.getLogger(__name__)
+
+
+def make_backward_function(
+        module: 'daceml.pytorch.DaceModule',
+        model: ONNXModel,
+        apply_strict=False) -> Type[torch.autograd.Function]:
+    """ Convert an ONNXModel to a PyTorch differentiable function. This method should not be used on it's own.
+        Instead use the ``backward=True`` parameter of :class:`daceml.pytorch.DaceModule`.
+
+        :param module: the parent pytorch module.
+        :param model: the model to convert.
+        :param apply_strict: whether to apply strict transformations before creating the backward pass.
+        :return: the PyTorch compatible :class:`torch.autograd.Function`.
+    """
+
+    if len(model.sdfg.nodes()) != 1:
+        raise AutoDiffException(
+            "Expected to find exactly one SDFGState, found {}".format(
+                len(model.sdfg.nodes())))
+
+    forward_sdfg = model.sdfg
+    forward_state = model.sdfg.nodes()[0]
+
+    backward_sdfg = dace.SDFG(forward_sdfg.name + "_backward")
+    backward_state = backward_sdfg.add_state()
+
+    gen = BackwardPassGenerator(
+        sdfg=forward_sdfg,
+        state=forward_state,
+        given_gradients=[clean_onnx_name(name) for name in model.outputs],
+        # TODO filter inputs that don't require grad somehow...
```

Comment:
you have access to the module, maybe named_parameters() can be used?

Reply:
Actually I don't think you can really decide this at this time, because you only know which inputs require grad at the moment the function is called (since you only get access to the inputs then)